lineapy/plugins/pipeline_writers.py

```diff
@@ -0,0 +1,208 @@
+import logging
+from pathlib import Path
+from typing import Dict, List, Optional
+
+from lineapy.data.types import PipelineType
+from lineapy.graph_reader.artifact_collection import ArtifactCollection
+from lineapy.graph_reader.session_artifacts import SessionArtifacts
+from lineapy.plugins.task import (
+    AirflowDagConfig,
+    AirflowDagFlavor,
+    TaskGraph,
+    TaskGraphEdge,
+)
+from lineapy.plugins.utils import load_plugin_template
+from lineapy.utils.logging_config import configure_logging
+from lineapy.utils.utils import get_system_python_version, prettify
+
+logger = logging.getLogger(__name__)
+configure_logging()
+
+
+def generate_pipeline_files(
+    artifact_collection: ArtifactCollection,
+    framework: str = "SCRIPT",
```

Comment:
discussed with @yoonspark , move to pipeline object.

Reply:
There is actually another ticker LIN-478 to integrate with Pipeline class.

Reply:
Will do this (using `ArtifactCollection` and pipeline writer objects in `Pipeline` object) in LIN-478. Should be straightforward and won't need 5 points.